fvtest/compilerunittest/p/BinaryEncoder.cpp

```diff
@@ -0,0 +1,2856 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "../CodeGenTest.hpp"
+#include "codegen/GenerateInstructions.hpp"
+#include "il/LabelSymbol.hpp"
+
+#include <iostream>
+#include <stdexcept>
+
+class BinaryInstruction {
```

Comment:
I'm not sure how useful this class would be outside of the Power codegen in its current form. At best, it could really only be used for RISC architectures due to its assumption that instructions are collections of words (while they're collections of bytes on x86, z, and other CISC architectures). On top of that, I'm not sure that the reasons that this class is needed in the first place even apply to other architectures. Those reasons being the ability to implicitly convert from a single `uint32_t` to reduce boilerplate in the most common case and to use the ^ operator to XOR instructions, which is only needed to deal with record form instructions.

I'm not going to rule out the possibility of generalizing this later, but I don't really know enough about what the other codegens would want to be able to generalize this at the moment.

Reply:
Fair enough. We can re-evaluate this when we get tests for the other binary encoders.